templates/lxc-local.in

```diff
@@ -180,6 +180,75 @@ process_excludes() {
   fi
 }
 
+extract_config() {
+  # lxc-create will automatically create a config file at ${LXC_PATH}/config.
+  # This function extracts the network config, and any remaining lxc config
+  # The extracted config is stored in ${LXC_PATH}/config-network and ${LXC_PATH}/config-extra
+  # The config will be merged later.
+
+  # Extract all the network config entries
+  sed -i -e "/lxc.net.0/{w ${LXC_PATH}/config-network" -e "d}" "${LXC_PATH}/config"
+
+  if [ -e "${LXC_PATH}/config-network" ]; then
+    echo "Extracted network config to: ${LXC_PATH}/config-network"
+    cat "${LXC_PATH}/config-network"
+  fi
+  
+  # Extract any other config entry
+  sed -i -e "/lxc./{w ${LXC_PATH}/config-extra" -e "d}" "${LXC_PATH}/config"
+
+  if [ -e "${LXC_PATH}/config-extra" ]; then
+    echo "Extracted additional config to: ${LXC_PATH}/config-extra"
+    cat "${LXC_PATH}/config-extra"
+  fi
+}
+
+add_container_config() {
+  # Adds the contents of the supplied metadata config to the container config
+  {
+    echo ""
+    echo "# Distribution configuration"
+    cat "$configfile"
+  } >> "${LXC_PATH}/config"
+}
+
+add_extra_config() {
+  # Add the container-specific config
+  {
+    echo ""
+    echo "# Container specific configuration"
+    if [ -e "${LXC_PATH}/config-extra" ]; then
+      cat "${LXC_PATH}/config-extra"
+      rm "${LXC_PATH}/config-extra"
+    fi
+  } >> "${LXC_PATH}/config"
+}
+
+add_network_config() {
+  # Re-add the previously removed network config
+  if [ -e "${LXC_PATH}/config-network" ]; then
+    {
+      echo ""
+      echo "# Network configuration"
+      cat "${LXC_PATH}/config-network"
+      rm "${LXC_PATH}/config-network"
+    } >> "${LXC_PATH}/config"
+  fi
+}
+
+process_config() {
+  # Process the supplied config file if it exists.
+  configfile="$(relevant_file config)"
+  if [ ! -e "${configfile}" ]; then
+    echo "ERROR: metadata tarball is missing the configuration file" 1>&2
```

Comment:
You can avoid a level of indentation here by just calling `return` after this error, avoiding having to use an `else` statement entirely.

Reply:
Thanks, I've implemented this change